sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_deserialize.py

```diff
@@ -69,56 +78,58 @@ def deserialize_file_properties(response, obj, headers):
     return deserialized.paths if deserialized.paths else {}, normalize_headers(response_headers)
 
 
-def get_deleted_path_properties_from_generated_code(generated):  # pylint: disable=name-too-long
+def get_deleted_path_properties_from_generated_code(generated: "BlobItemInternal") -> DeletedPathProperties:  # pylint: disable=name-too-long
     deleted_path = DeletedPathProperties()
     deleted_path.name = generated.name
-    deleted_path.deleted_time = generated.properties.deleted_time
-    deleted_path.remaining_retention_days = generated.properties.remaining_retention_days
-    deleted_path.deletion_id = generated.deletion_id
+    deleted_path.deleted_time = generated.properties.deleted_time  # type: ignore [assignment]
+    deleted_path.remaining_retention_days = generated.properties.remaining_retention_days  # type: ignore [assignment]
+    deleted_path.deletion_id = generated.deletion_id  # type: ignore [assignment]
     return deleted_path
 
 
-def is_file_path(_, __, headers):
-    if headers['x-ms-resource-type'] == "file":
-        return True
-    return False
+def is_file_path(_, __, headers: Dict[str, Any]) -> bool:
+    return headers['x-ms-resource-type'] == "file"
 
 
-def get_datalake_service_properties(datalake_properties):
-    datalake_properties["analytics_logging"] = AnalyticsLogging._from_generated(    # pylint: disable=protected-access
+def get_datalake_service_properties(datalake_properties: Dict[str, Any]) -> Dict[str, Any]:
+    datalake_properties["analytics_logging"] = AnalyticsLogging._from_generated(  # pylint: disable=protected-access
         datalake_properties["analytics_logging"])
     datalake_properties["hour_metrics"] = Metrics._from_generated(datalake_properties["hour_metrics"])  # pylint: disable=protected-access
-    datalake_properties["minute_metrics"] = Metrics._from_generated(    # pylint: disable=protected-access
+    datalake_properties["minute_metrics"] = Metrics._from_generated(  # pylint: disable=protected-access
         datalake_properties["minute_metrics"])
-    datalake_properties["delete_retention_policy"] = RetentionPolicy._from_generated(   # pylint: disable=protected-access
+    datalake_properties["delete_retention_policy"] = RetentionPolicy._from_generated(  # pylint: disable=protected-access
         datalake_properties["delete_retention_policy"])
     datalake_properties["static_website"] = StaticWebsite._from_generated(  # pylint: disable=protected-access
         datalake_properties["static_website"])
     return datalake_properties
 
 
-def from_blob_properties(blob_properties, **additional_args):
+def from_blob_properties(blob_properties: "BlobProperties", **additional_args: Any) -> FileProperties:
     file_props = FileProperties()
     file_props.name = blob_properties.name
     file_props.etag = blob_properties.etag
-    file_props.deleted = blob_properties.deleted
+    file_props.deleted = blob_properties.deleted  # type: ignore [assignment]
```

Comment:
Worth casting from `BlobProperties` to `FileProperties`? Or just leave this as is?